dio/lib/src/options.dart

```diff
@@ -239,32 +242,36 @@ class RequestOptions extends Options {
           maxRedirects: maxRedirects,
           requestEncoder: requestEncoder,
           responseDecoder: responseDecoder,
-        );
+        ) {
+    this.path = path ?? this.path;
+    this.queryParameters = queryParameters ?? {};
```

Comment:
Again, queryParameters can be in the initializer list and doesn't need to be late.